hbase/hbase/stackable/patches/2.6.1/0006-Update-property-usage-for-bound-ports.patch

```diff
@@ -0,0 +1,174 @@
++import static org.apache.hadoop.hbase.HConstants.MASTER_INFO_PORT;
+ import static org.apache.hadoop.hbase.HConstants.MASTER_PORT;
+ import static org.apache.hadoop.hbase.master.cleaner.HFileCleaner.CUSTOM_POOL_SIZE;
+ import static org.apache.hadoop.hbase.util.DNS.MASTER_HOSTNAME_KEY;
+@@ -569,7 +569,7 @@ public class HMaster extends HRegionServer implements MasterServices {
+ 
+   @Override
+   protected int getUseThisInfoPortInstead(Configuration conf) {
+-    int port = conf.getInt(MASTER_BOUND_INFO_PORT, 0);
++    int port = conf.getInt(MASTER_INFO_PORT, 0);
+     return port != 0 ? port : this.infoServer != null ? this.infoServer.getPort() : -1;
+   }
+ 
+@@ -3143,7 +3143,7 @@ public class HMaster extends HRegionServer implements MasterServices {
+   public int getRegionServerInfoPort(final ServerName sn) {
+     int port = this.serverManager.getInfoPort(sn);
+     return port == 0
+-      ? conf.getInt(HConstants.REGIONSERVER_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT)
++      ? conf.getInt(HConstants.REGIONSERVER_BOUND_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT)
+       : port;
+   }
+ 
+diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+index 11bd1e58b5..abeeab3552 100644
+--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
++++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+@@ -25,6 +25,7 @@ import static org.apache.hadoop.hbase.HConstants.DEFAULT_SLOW_LOG_SYS_TABLE_CHOR
+ import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_COORDINATED_BY_ZK;
+ import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_MAX_SPLITTER;
+ import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_BOUND_INFO_PORT;
++import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_INFO_PORT;
+ import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_PORT;
+ import static org.apache.hadoop.hbase.HConstants.RPC_CLIENT_BIND_ADDRESS;
+ import static org.apache.hadoop.hbase.master.waleventtracker.WALEventTrackerTableCreator.WAL_EVENT_TRACKER_ENABLED_DEFAULT;
+@@ -791,7 +792,7 @@ public class HRegionServer extends Thread
+   }
+ 
+   protected int getUseThisInfoPortInstead(Configuration conf) {
+-    int port = conf.getInt(REGIONSERVER_BOUND_INFO_PORT, 0);
++    int port = conf.getInt(REGIONSERVER_INFO_PORT, 0);
+     return port != 0 ? port : this.infoServer != null ? this.infoServer.getPort() : -1;
+   }
+ 
+@@ -2463,11 +2464,11 @@ public class HRegionServer extends Thread
+    */
+   private void putUpWebUI() throws IOException {
+     int port =
+-      this.conf.getInt(HConstants.REGIONSERVER_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
++      this.conf.getInt(REGIONSERVER_BOUND_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
```

Comment:
This has to default to the advertised port, so along the lines of `getInt(REGIONSERVER_BOUND_INFO_PORT, getInt(REGIONSERVER_INFO_PORT, DEFAULT_REGIONSERVER_INFOPORT))`.

Reply:
65aaae2